test/extensions/io_socket/buffered_io_socket/buffered_io_socket_handle_impl_test.cc

```diff
@@ -245,6 +245,59 @@ TEST_F(BufferedIoSocketHandleTest, FlowControl) {
   EXPECT_TRUE(io_handle_->isWritable());
 }
 
+// Consistent with other IoHandle: allow write empty data when handle is closed.
```

Comment:
This seems weird, should probably change both implementations, but that's for me to do.